librz/arch/function.c

```diff
@@ -282,6 +297,38 @@ RZ_API bool rz_analysis_function_rename(RzAnalysisFunction *fcn, const char *nam
 	return true;
 }
 
+/**
+ * @b Force rename a function.
+ * This will make sure a function with non-unique name gets renamed by
+ * adding a numeric suffix to it.
+ *
+ * @param fcn
+ * @param name
+ *
+ * @return New name on success.
+ * @return `NULL` otherwise.
+ *
+ * @sa rz_analysis_function_rename
+ * */
+RZ_API const char *rz_analysis_function_force_rename(RzAnalysisFunction *fcn, const char *name) {
+	rz_return_val_if_fail(fcn && name, NULL);
+
+	const char *final_name = strdup(name);
+	ut32 retries = 100;
+	ut32 suffix = 0;
+	while ((retries--) && !rz_analysis_function_rename(fcn, final_name)) {
+		if (final_name) {
+			free((void *)final_name);
+			final_name = NULL;
+		}
+
+		ut32 strsz = snprintf(NULL, 0, "%s_%u", name, suffix) + 1;
+		final_name = malloc(strsz);
+		snprintf((char *)final_name, strsz, "%s_%u", name, suffix++);
+	}
+	return retries ? fcn->name : NULL;
```

Comment:
instead of doing retries, just do `name_offset` which is granted to be unique. also use `rz_strf` directly instead of calling snprintf.